build/common.bzl

```diff
@@ -50,26 +50,48 @@ EXECUTABLE_FLAGS = [
 # Now we start declaring macros to help us with common patterns
 # such as pulling tree-sitter grammars from releases/git hashes.
 
-def tree_sitter_node_types_release(name, version, sha256):
-    """Create a package for a tree-sitter grammar and export its node-types.json file/test corpus.."""
+_tree_sitter_language_build = """
+package(default_visibility = ["//visibility:public"])
+
+load("@rules_haskell//haskell:cabal.bzl", "haskell_cabal_library")
+load("@stackage//:packages.bzl", "packages")
+exports_files(glob(["**/node-types.json"]))
+
+alias(
+   name = "src/node-types.json",
+   actual = "{node_types_path}",
+)
+
+haskell_cabal_library(
+    name = "{name}",
+    version = "{version}",
+    srcs = glob(["**"]),
+    deps = packages["{name}"].deps,
+    visibility = ["//visibility:public"],
+)
+
+filegroup(name = "corpus", srcs = glob(["**/corpus/*.txt"]))
```

Comment:
This string stands in for the role played by `BUILD.bazel` files in the languages’ packages. Since we have to parameterize it according to the arguments passed to `tree_sitter_node_types_hackage`, we use Python’s `{format_strings}`.